core/feconf.py

```diff
@@ -111,7 +111,7 @@
 # algorithms per interaction. Hence, whenever we find a secondary algorithm
 # candidate for any of the supported interactions, the logical functions to
 # support multiple algorithms need to be implemented.
-INTERACTION_CLASSIFIER_MAPPING = {
+INTERACTION_CLASSIFIER_MAPPING: Dict[str, Dict[str, Union[str, int]]] = {
```

Comment:
Actually the interior items should probably be typed as a named dict instead -- i.e. every key in INTERACTION_CLASSIFIER_MAPPING will have a value which is a dict with two keys, algorithm_id and algorithm_version. The former will map to a string and the latter to an int.

Reply:
I tried to do it. Please have a look.

Reply:
I tried to add named dict by [this](e37dfa85a25bff03787c45411aaf09de4cc6c052) commit. But I don't know why the tests are failing.

Reply:
I discussed it with vojtech jelinek and he proposed to update `install_third_party_libs.py` with `typing_extension`. I will try to do this and update the PR soon.

Reply:
I have updated `install_third_party_libs.py` and the corresponding test file. Please have a look and let me know if any changes are needed.